submission/submitter/src/main/kotlin/ac/uk/ebi/biostd/submission/submitter/request/SubmissionRequestLoader.kt

```diff
@@ -1,25 +1,39 @@
-package ac.uk.ebi.biostd.submission.submitter
+package ac.uk.ebi.biostd.submission.submitter.request
 
 import ac.uk.ebi.biostd.persistence.common.request.SubmissionRequest
 import ac.uk.ebi.biostd.persistence.common.service.SubmissionPersistenceQueryService
 import ac.uk.ebi.biostd.persistence.filesystem.service.FileProcessingService
 import ebi.ac.uk.extended.model.ExtFile
+import ebi.ac.uk.extended.model.ExtSubmission
 import ebi.ac.uk.extended.model.FireFile
 import ebi.ac.uk.extended.model.NfsFile
 import ebi.ac.uk.io.ext.md5
 import ebi.ac.uk.io.ext.size
+import mu.KotlinLogging
 
-class SubmissionRequestProcessor(
+private val logger = KotlinLogging.logger {}
+
+class SubmissionRequestLoader(
     private val submissionPersistenceQueryService: SubmissionPersistenceQueryService,
     private val fileProcessingService: FileProcessingService,
 ) {
 
     internal fun loadRequest(accNo: String, version: Int): SubmissionRequest {
+        logger.info { "Loading request accNo='$accNo', version='$version'" }
         val rqt = submissionPersistenceQueryService.getPendingRequest(accNo, version)
-        val full = fileProcessingService.processFiles(rqt.submission) { loadFileAttributes(it) }
+        val full = processRequest(rqt.submission)
+        logger.info { "Finish Loading request accNo='$accNo', version='$version'" }
         return SubmissionRequest(full, rqt.fileMode, rqt.draftKey)
     }
 
+    private fun processRequest(sub: ExtSubmission): ExtSubmission {
+        var filesCount = 0
+        return fileProcessingService.processFiles(sub) {
```

Comment:
Do you think here we could calculate the whole size as well?

Reply:
yes here is probably the best moment to do it. When request is loaded. Clearly out of the scope of this ticket.